extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchToQuantilesSketchPostAggregator.java

```diff
@@ -0,0 +1,150 @@
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.tuple;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.druid.query.aggregation.AggregatorUtil;
+import io.druid.query.aggregation.PostAggregator;
+
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketch;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketchIterator;
+import com.yahoo.sketches.quantiles.DoublesSketch;
+import com.yahoo.sketches.quantiles.UpdateDoublesSketch;
+
+/**
+ * Returns a quanitles {@link DoublesSketch} constructed from a given column of double values from a given
+ * {@link ArrayOfDoublesSketch} using parameter k that determines the accuracy and size of the quantiles sketch.
+ * The column number is optional (the default is 1).
+ * The parameter k is optional (the default is defined in the sketch library).
+ * The result is a quantiles sketch.
+ * See <a href=https://datasketches.github.io/docs/Quantiles/QuantilesOverview.html>Quantiles Sketch Overview</a>
+ */
+public class ArrayOfDoublesSketchToQuantilesSketchPostAggregator extends ArrayOfDoublesSketchUnaryPostAggregator
+{
+
+  private static final int DEFAULT_QUANTILES_SKETCH_SIZE = 128;
+
+  private final int column;
+  private final int k;
+
+  @JsonCreator
+  public ArrayOfDoublesSketchToQuantilesSketchPostAggregator(
+      @JsonProperty("name") final String name,
+      @JsonProperty("field") final PostAggregator field,
+      @JsonProperty("column") @Nullable final Integer column,
+      @JsonProperty("k") @Nullable final Integer k
+  )
+  {
+    super(name, field);
+    this.column = column == null ? 1 : column;
+    this.k = k == null ? DEFAULT_QUANTILES_SKETCH_SIZE : k;
+  }
+
+  @Override
+  public Comparator<DoublesSketch> getComparator()
+  {
+    return new Comparator<DoublesSketch>()
+    {
+      @Override
+      public int compare(final DoublesSketch a, final DoublesSketch b)
+      {
+        return Long.compare(a.getN(), b.getN());
+      }
+    };
+  }
+
+  @Override
+  public Object compute(final Map<String, Object> combinedAggregators)
```

Comment:
Please change the return type to `DoublesSketch` for type safety.

Reply:
ok